apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts

```diff
@@ -1,58 +1,302 @@
-        {
-          provide: getRepositoryToken(POSDepositEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: getRepositoryToken(LocationEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: TransactionService,
-          useValue: mockedTransactionService
-        },
-        {
-          provide: PaymentService,
-          useValue: mockedPaymentService
-        }
-      ]
-    }).compile();
+      providers: [POSReconciliationService]
+    })
+      .useMocker(createMock)
+      .compile();
+
+    /**
+     * Generate mock data and update some of the payment to not match for round one time heuristic
+     */
+    const data = new MockData(PaymentMethodClassification.POS);
+
+    payments = data.paymentsMock.map((itm: PaymentEntity, index: number) =>
+      setSomePaymentsToTwentyMinutesLater(itm, index)
+    ) as PaymentEntity[];
+
+    deposits = data.depositsMock as POSDepositEntity[];
 
     service = module.get<POSReconciliationService>(POSReconciliationService);
+    posDepositService = module.get(PosDepositService);
+    paymentService = module.get(PaymentService);
+    logger = module.get(Logger);
+    service.verifyMatch = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('initialization', () => {
+    it('should create the service and its dependencies', () => {
+      expect(service).toBeDefined();
+      expect(posDepositService).toBeDefined();
+      expect(paymentService).toBeDefined();
+      expect(logger).toBeDefined();
+    });
+  });
+
+  describe('matchPaymentsToDeposits first round heuristics', () => {
+    it('should verify the test data status is PENDING prior to running reconciliation', () => {
+      payments.forEach((itm) => expect(itm.status).toBe(MatchStatus.PENDING));
+      deposits.forEach((itm) => expect(itm.status).toBe(MatchStatus.PENDING));
+    });
+
+    it('calls a function to match based on the first round of heuristics', () => {
+      const timeDiff = 5;
+      const spy = jest.spyOn(service, 'matchPosPaymentToPosDeposits');
+
+      matchedRoundOne = service.matchPosPaymentToPosDeposits(
+        payments,
+        deposits,
+        timeDiff
+      );
+
+      expect(spy).toHaveBeenCalledTimes(1);
+      /**
+       * Test that calling matchPosPaymentToDeposits calls verifyMatch
+       */
+      expect(service.verifyMatch).toBeCalled();
+    });
+
+    it('should verify the time difference between matches on round one is 5 minutes or less', () => {
+      matchedRoundOne.forEach(({ payment, deposit }) =>
+        expect(
+          timeBetweenMatchedPaymentAndDeposit(payment, deposit)
+        ).toBeLessThanOrEqual(5)
+      );
+    });
+
+    it('should have updated the status from PENDING to MATCH', () => {
+      matchedRoundOne.forEach((itm) => {
+        expect(itm.deposit.status === MatchStatus.MATCH);
+        expect(itm.payment.status === MatchStatus.MATCH);
+      });
+    });
```

Comment:
Do we know exactly how many matches we have, and how many non-matches we have? Is there an argument for adding that as a test too?